server/lambda/index.js

```diff
@@ -0,0 +1,325 @@
+        telephone: 1,
+        id: 1,
+        derivedTravellerKey: 1,
+        address: 1,
+        dob: 1,
+        traveller0Dob: 1,
+        traveller1Dob: 1,
+        traveller2Dob: 1,
+        traveller3Dob: 1,
+        traveller4Dob: 1,
+        traveller5Dob: 1,
+        traveller6Dob: 1,
+        traveller7Dob: 1,
+        traveller8Dob: 1,
+        traveller9Dob: 1,
+        arrivalDate: 1,
+        isoAddress: 1,
+      },
+    },
+  ]).toArray();
+
+  // Maps to store keys representing different aspects of the entries from ETS and PHAC
+  const duplicates = [];
+  const etsKeys = createEtsKeys(ets);
+  const phacKeys = [];
+
+  let internalPhacDuplicateCount = 0;
+  let phacToEtsDuplicateCount = 0;
+
+  // Iterate through the PHAC collection
+  for (const phacEntry of phac) {
+    // Load entry data
+    const covidId = phacEntry.covid_id;
+    const phacArrivalDate = phacEntry.arrival_date;
+    const phacEntryKeys = [];
+    const phacPhones = [
+      cleanString(phacEntry.home_phone),
+      cleanString(phacEntry.mobile_phone),
+      cleanString(phacEntry.other_phone),
+    ];
+    // Create keys for the entry
+    phacPhones.forEach((item) => { if (item) { phacEntryKeys.push(item); } });
+    phacEntryKeys.push(phacEntry.derivedTravellerKey);
+    phacEntryKeys.push(`${transformAddress(phacEntry.address_1)}${cleanString(phacEntry.date_of_birth)}`);
+
+    // Iterate through the keys to compare against our collections
+    for (const phacKey of phacEntryKeys) {
+      // Check if PHAC already contains any of our keys, if not then add them
+      if (phacKeys[phacKey] && phacKeys[phacKey] !== covidId) {
+        // Match found, compare dates to see if within a week or not
+        if (compareDates(
+          phacArrivalDate, phac.find((item) => item.covid_id === covidId).arrival_date, 7,
+        )) {
+          if (duplicates[covidId]) {
+            duplicates[covidId] = merge(duplicates[covidId], phacKeys[phacKey]);
+          } else {
+            duplicates[covidId] = flatten([phacKeys[phacKey]]);
+            internalPhacDuplicateCount += 1;
+          }
+        }
+      } else {
+        phacKeys[phacKey] = covidId;
+      }
+
+      // Check if ETS contains any of our keys
+      if (etsKeys[phacKey]) {
+        // Extract dates
+        const etsDates = [];
+
+        for (const etsCode of etsKeys[phacKey]) {
+          etsDates.push(ets.find((item) => item.id === etsCode).arrivalDate);
+        }
+
+        // If not within a date of any ETS match, treat as unique
+        const insideRange = etsDates.find((item) => compareDates(phacArrivalDate, item, 7));
+
+        if (insideRange) {
+          if (duplicates[covidId]) {
+            duplicates[covidId] = merge(duplicates[covidId], etsKeys[phacKey]);
+          } else {
+            duplicates[covidId] = flatten([etsKeys[phacKey]]);
+            phacToEtsDuplicateCount += 1;
+          }
+        }
+      }
+    }
+  }
+
+  /* eslint-disable no-console */
```

Comment:
Lots of commented out code here...

Reply:
Discussed offline - plan is to remove the per-record details as these will be captured in the DB directly in a future PR, and re-enable the output summarizing the results so this will be captured in CloudWatch logs